### ISSUE TITLE
 Use script global line to display debug info alternatively?

### ISSUE THREAD

**Issue by syswow** [neovim/neovim#2565]
Current Neovim's debug info is the same with Vim,  the line start from the definition line of func.
eg:
```
Error detected while processing function ABC..<SNR>21_ABC..<SNR>21_ABC:
line   21:
E121: Undefined variable: abc
```
Would Neovim use script global line to display debug info alternatively?

**Comment by justinmk** [comment]
I don't understand the request.

**Comment by syswow** [comment]
@justinmk  Sorry for my poor English, I have improved the request. Please check again. :smile:

**Comment by justinmk** [comment]
Why would you want that?

**Comment by kopischke** [comment]
@justinmk at a guess, I would say to make jumps to the troublesome line easier: right now, it’s a file, location, offset jump where it could be a file, line jump. I agree the current state adds some friction to debugging Vim scripts.

**Comment by ZyX-I** [comment]
@justinmk Currently you need to do the following:

0. Deduce which file function Test is in (maybe using `:verbose function Test`.
1. Find function Test in a file (may be troublesome if it is anonymous function).
2. Make sure that there are no folds inside the function (even with `set foldopen=all` `5j` will jump over the fold).
3. Use `{line}j` to jump onto the offending line.

With the proposed change this will be

0. Deduce which file function Test is in.
1. Use `{line}gg`. Possibly open the fold afterwards (not needed in case `set foldopen+=jump`).

**Comment by justinmk** [comment]
This is a potentially breaking change, but if others are in favor of it I do agree it makes sense. I do not think it should be optional though, we should just change it. Any objections to that?

**Comment by syswow** [comment]
Thanks to you all for the comments.

> Why would you want that?

IMO, using absolute line in VimL's debug info can make jumping onto bug-line and debugging more convenient.

**Comment by jalvesaq** [comment]
The current error message has the advantage of telling the name of the function where the bug happened. Why not add the name of the file and its line in parenthesis? Example:

```
Error detected while processing function Test, line 1 (script.vim, line 3):
```

**Comment by jalvesaq** [comment]
The expression "detected while processing"  is redundant. The message could be:
```
Error in function Test, line 1 (script.vim, line 3):
```

**Comment by justinmk** [comment]
@jalvesaq Good ideas. How about this:

    Error: script.vim(line 3): in foo#bar#Test(line 1):

**Comment by jalvesaq** [comment]
@justinmk I agree about the top down approach. I did not like the 3 colons in the "sentence", but this seems to be usual in programming. GCC reports errors in a similar way.
